engine/particles/private/particle_module.cpp

```diff
@@ -1,17 +1,177 @@
 #include "particle_module.hpp"
+
+#include "components/rigidbody_component.hpp"
+#include "components/transform_component.hpp"
+#include "components/transform_helpers.hpp"
+#include "components/world_matrix_component.hpp"
 #include "ecs_module.hpp"
+#include "emitter_component.hpp"
 #include "engine.hpp"
-#include "particle_interface.hpp"
+#include "gpu_resources.hpp"
+#include "graphics_context.hpp"
+#include "graphics_resources.hpp"
+#include "physics_module.hpp"
 #include "renderer.hpp"
-#include "renderer_module.hpp"
-
-#include <memory>
+#include "resource_management/image_resource_manager.hpp"
+#include "time_module.hpp"
+#include <renderer_module.hpp>
 
 ModuleTickOrder ParticleModule::Init(Engine& engine)
 {
-    auto& ecs = engine.GetModule<ECSModule>();
-    auto renderer = engine.GetModule<RendererModule>().GetRenderer();
-    _particleInterface = std::make_unique<ParticleInterface>(renderer->GetContext(), ecs);
+    _physics = &engine.GetModule<PhysicsModule>();
+    _context = engine.GetModule<RendererModule>().GetRenderer()->GetContext();
+    _ecs = &engine.GetModule<ECSModule>();
+
+    return ModuleTickOrder::ePostRender;
+}
+
+void ParticleModule::Tick(MAYBE_UNUSED Engine& engine)
+{
+    const auto emitterView = _ecs->GetRegistry().view<EmitterComponent>();
+    for (const auto entity : emitterView)
+    {
+        if (_ecs->GetRegistry().all_of<RigidbodyComponent>(entity))
```

Comment:
maybe i dont understand `all_of` but why not add Rigidbodycomponent to the view?